### PR TITLE
bin: allow wallet to run separately from node server

### PR DESCRIPTION
**Pull request by pinheadmz**
This is a cherry-pick of https://github.com/bcoin-org/bcoin/pull/652
(fullnode already parses `--no-wallet`)

...which also relies on a needed backport from bcoin: https://github.com/bcoin-org/bcoin/commit/d25e553605eaa8e8a55a04aacec0e238bc0e111d

and the more subtle: https://github.com/bcoin-org/bcoin/commit/2efa5275d53c8f2efbb4c1af638ce9ac85889520

...which are included in this PR.

Launch hsd full or spv with no wallet plugin:
`$ hsd --spv --no-wallet`

Install globally (after merging or pulling):
`npm install -g`

Launch wallet as separate server:
`$ hswallet --network=testnet`

Client commands work normally:
`$ hsw-cli --network=testnet send ts1qwdd3xdw54u2mfdnltz6lnt4vr8ks5lc3seexjc 0.1`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,7 +60,8 @@
   "bin": {
     "hsd": "./bin/hsd",
     "hsd-node": "./bin/node",
-    "hsd-spvnode": "./bin/spvnode"
+    "hsd-spvnode": "./bin/spvnode",
+    "hswallet": "./bin/hsw"
```
Comment (boymanjor) [inline package.json]: nit: in order to follow the naming conventions of other libraries, can you change this binary to `hs-wallet`?

Comment (pinheadmz) [reply package.json]: done! thanks.

**Comment by boymanjor** [comment]
@pinheadmz this looks good to merge. Do you mind changing your commit message to match the title of this PR?

**Comment by pinheadmz** [comment]
@boymanjor squashed, rebased, renamed 🚀

**Pull request merged**